generator/transit_generator.cpp

```diff
@@ -111,17 +129,71 @@ struct IsSortedVisitor
   bool m_isSorted = true;
 };
 
-/// \returns ref to a stop at |stops| by |stopId|.
-Stop const & FindStopById(vector<Stop> const & stops, StopId stopId)
+template <typename T>
+void Append(vector<T> const & src, vector<T> & dst)
+{
+  dst.insert(dst.end(), src.begin(), src.end());
+}
+
+void LoadBorders(string const & dir, TCountryId const & countryId, vector<m2::RegionD> & borders)
+{
+  string const polyFile = my::JoinPath(dir, BORDERS_DIR, countryId + BORDERS_EXTENSION);
+  borders.clear();
+  osm::LoadBorders(polyFile, borders);
+}
+
+bool HasStop(vector<Stop> const & stops, StopId stopId)
+{
+  return binary_search(stops.cbegin(), stops.cend(), Stop(stopId));
+}
+
+/// \brief Removes from |items| all items which stop ids is not contained in |stops|.
+/// \note This method keeps relative order of |items|.
+template <class Item>
+void ClipItemsByStops(vector<Stop> const & stops, vector<Item> & items)
+{
+  CHECK(is_sorted(stops.cbegin(), stops.cend()), ());
```

Comment:
As you checks that stops are sorted, and then iterates over items, you may simply "merge" items with stops, which reduces complexity from O(m + nlogm) to O(m + n). Up to you, of course.

Reply:
I didn't I get your idea. What do you mean by saying '"merge" items with stops'? (|items| and |stops| have different types).

Reply:
Sorry, forget about that.